lib/jekyll/cache.rb

```diff
@@ -12,23 +12,34 @@ class Cache
 
     class << self
       # class-wide cache location
-      attr_accessor :cache_dir
+      attr_reader :cache_dir
 
       # class-wide directive to write cache to disk
       attr_reader :disk_cache_enabled
 
       # class-wide base cache reader
       attr_reader :base_cache
 
+      # Setup cache attributes for given +site+
+      #
+      # site - A Jekyll::Site instance
+      #
+      # Returns nothing
+      def bootstrap(site)
+        @cache_dir = determine_cache_dir(site)
+        disable_disk_cache! if @cache_dir.nil? || site.safe
+        nil
+      end
+
       # Disable Marshaling cached items to disk
       def disable_disk_cache!
         @disk_cache_enabled = false
       end
 
-      # Clear all caches
+      # Delete all cached items from the disk cache and clear all in-memory caches
       def clear
-        delete_cache_files
-        base_cache.each_value(&:clear)
+        FileUtils.rm_rf(cache_dir) if disk_cache_enabled && cache_dir
```

Comment:
```suggestion
        FileUtils.rm_rf(cache_dir) if disk_cache_enabled?
```

Reply:
This won't do because `:clear` is a class method and `:disk_cache_enabled?` an instance_method..